clang/tools/driver/cc1_main.cpp

```diff
@@ -191,9 +191,17 @@ static int PrintSupportedCPUs(std::string TargetStr) {
 
 static Optional<llvm::cas::CASID>
 createResultCacheKey(llvm::cas::CASDB &CAS, DiagnosticsEngine &Diags,
-                     StringRef RootIDString, ArrayRef<const char *> Argv) {
+                     const CompilerInvocation &Invocation) {
+  // Generate a new command-line in case Invocation has been canonicalized.
+  llvm::BumpPtrAllocator Alloc;
+  llvm::StringSaver Saver(Alloc);
+  llvm::SmallVector<const char *> Argv;
+  Invocation.generateCC1CommandLine(
+      Argv, [&Saver](const llvm::Twine &T) { return Saver.save(T).data(); });
+
```

Comment:
I don't think this will affect performance for cache hits, but it certainly could. I'll check.

Here is the "before" test (which is some old toolchain I happen to have installed), and this is what I get for `time ninja clang`:
```
[3606/3606] Creating executable symlink bin/clang
ninja clang  54.55s user 69.76s system 791% cpu 15.704 total
```

I'll `ninja install-clang` and run an "after" test.

Reply:
The "spot check" looks okay: `ninja SupportTests` is still near-instant when cached. Waiting for the initial build of clang.